tutorials/profiler.md

```diff
@@ -99,7 +100,39 @@ xdg-open /usr/share/ignition/ignition-common3/profiler_vis/index.html
 open /usr/share/ignition/ignition-common3/profiler_vis/index.html
 ```
 
-If you are successful, you should see the profiler output in a browser.
+### On Ignition library
+
+If you want to use profiler on any other ignition library, enable the profiler at compile time with ``ENABLE_PROFILER`` cmake argument.
+
+When compiling with ``CMake``:
+```{.sh}
+cmake .. -DENABLE_PROFILER=1
+```
+When compiling with ``colcon``:
+```{.sh}
+colcon build --cmake-args -DENABLE_PROFILER=1
+```
+
+If the profiler is run successfully, you should see output in a browser. Similar to this 
+
+![profiler viewer exmaple](imgs/profiler_tutorial_example.png)
+
+### TroubleShoot the web viewer
+
+If you see ``connection error``, there are a couple of things to double check
+1. Was the profiler enabled when the project you're trying to run was compiled? Note that this isn't the case if you installed Ignition libraries from binaries, for example. You need to compile the project from source with the ENABLE_PROFILER variable set.
+2. Are you using the correct port number in the upper left corner ``Connection Addresss: ws://127.0.0.1:1500/rmt``? Run ``ign gazebo -v 4`` or ``echo $RMT_PORT`` to show the port number in use. The port number will be printed out if the profiler is enabled. 
```

Comment:
minor note here that ```echo $RMT_PORT``` will not print out anything if the user has not set this env variable. I think an easy change is to just leave out `or ``echo $RMT_PORT``` text. Point 4 should be sufficient to cover this port env usage.

Reply:
It should be almost always unset.